rocAL/rocAL/source/readers/image/coco_file_source_reader.cpp

```diff
@@ -86,9 +86,55 @@ Reader::Status COCOFileSourceReader::initialize(ReaderConfig desc)
             replicate_last_batch_to_pad_partial_shard();
         }
     }
-    //shuffle dataset if set
-    if (ret == Reader::Status::OK && _shuffle)
-        std::random_shuffle(_file_names.begin(), _file_names.end());
+
```

Comment:
I am seeing this same code is repeated twice in "reset()"  & "initialize".
Put this part of the code in a function block & call it when required